app/assets/javascripts/miq_application.js

```diff
@@ -692,8 +692,51 @@ function miqBuildChartMenu(col, row, value, category, series, id, message) {
   }
 }
 
+function miqChartBindEvents(chart_set, chart_index) {
+  if (ManageIQ.charts.provider == 'jqplot') {
+    jqplot_bind_events(chart_set, chart_index);
+  } else if (ManageIQ.charts.provider == 'c3') {
+    // noop
+  }
+}
+
+function miqBuildChartMenuEx(col, row, _value, category, series, chart_set, chart_index) {
+  var chart_data = ManageIQ.charts.chartData[chart_set];
+  var chart_el   = $('#miq_chart_parent_'+chart_set+'_'+chart_index);
+  var chartmenu_el = $('#miq_chartmenu_'+chart_set+'_'+chart_index);
+  chartmenu_el.empty();
+
+  if (chart_data[chart_index].menu != null && chart_data[chart_index].menu.length) {
+    var rowcolchart_index = "_" + row + "-" + col + "-" + chart_index;
+
+    for (var i = 0; i < chart_data[chart_index].menu.length; i++) {
```

Comment:
I think we may want to prefer `chart_data[chart_index].menu.forEach(function(menuItem) { ...` (seems like we're only ever using `i` to retrieve that item, not because we actually need the number..)